### PR TITLE
Add Per-Membership Rates to PMP Integration

### PR DESCRIPTION
**Pull request by codelion7**
Updated calculate_referral_amount()
	- Add $membership_level
	- Add logic for membership-specific rate type

Update affwp_calc_referral_amount()
	- Add $type and $membership_level
	- Add logic for membership-specific rate type

Add get_membership_rate()
	- Add alternate way of checking settings (get_option() instead of get_postmeta())

Update add_pending_referral() - PMP integration function
	- Add $order->membership_level to $referral_total calculation
	- Add check for disabled referrals on specific membership levels

Add new PMP integration functions
	- affwp_membership_settings()
	- save_membership_settings()

Resolves: #498

### CODE REVIEW

File: includes/integrations/class-base.php [hunk 1]
```diff
@@ -196,7 +196,7 @@ public function get_affiliate_email() {
 	 * @since   1.2
 	 * @return  array
 	 */
-	public function calculate_referral_amount( $base_amount = '', $reference = '', $product_id = 0 ) {
+	public function calculate_referral_amount( $base_amount = '', $reference = '', $product_id = 0, $membership_level = 0 ) {
```
Comment (pippinsplugins) [inline includes/integrations/class-base.php]: The `$member_level` here should be passed as the `$product_id`.

File: includes/integrations/class-pmp.php [hunk 2]
```diff
@@ -107,10 +118,103 @@ public function reference_link( $reference = 0, $referral ) {
...
+	 * @access  public
+	 * @since   1.5.7
+	*/
+	public function affwp_membership_settings() {
+	
+		$level_id = intval($_REQUEST['edit']);
+		
+		if( ! $level_id > 0 ) return;
+
+		$rate = get_option( "affwp_membership_rate_" . $level_id );
```
Comment (pippinsplugins) [inline includes/integrations/class-pmp.php]: I would like for these option keys to include `affwp_pmp_` in the name so it's clear they're related to Paid Memberships Pro.

File: includes/integrations/class-pmp.php [hunk 3]
```diff
@@ -107,10 +118,103 @@ public function reference_link( $reference = 0, $referral ) {
...
+		$type = get_option( "affwp_membership_rate_type_" . $level_id );
+		$disabled = get_option( "affwp_membership_referrals_disabled_" . $level_id );
+		
+		if( ! $rate ) $rate = "";
```
Comment (pippinsplugins) [inline includes/integrations/class-pmp.php]: Please use `{}` for all conditionals in order to follow our coding standards.

**Comment by pippinsplugins** [comment]
Thanks for this @codelion7! I've added a couple of notes for you above on some minor items.

**Comment by pippinsplugins** [comment]
@codelion7 Could you resubmit this to the `release/1.6` branch?

**Comment by codelion7** [comment]
Alright, I've resubmitted my pull request #586 to the release/1.6 branch, as requested. I've made some improvements as well.

**Comment by pippinsplugins** [comment]
Awesome!

**Pull request closed without merging**